test/e2e-encryption-kms/encryption_kms.go

```
@@ -90,8 +95,75 @@ func testKMSEncryptionProvidersMigration(t testing.TB) {
+			TargetGRs:                       operatorencryption.DefaultTargetGRs,
+			AssertFunc:                      operatorencryption.AssertSecretsAndConfigMaps,
+		},
+		InvalidImageProvider: invalidProvider,
+		ValidImageProvider:   librarykms.DefaultFakeVaultEncryptionProvider,
+		WaitForDegraded: func(ctx context.Context, t testing.TB) {
+			t.Helper()
+			operatorClient := operatorencryption.GetOperator(t)
+			err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 10*time.Minute, true, func(ctx context.Context) (bool, error) {
+				operator, err := operatorClient.Get(ctx, "cluster", metav1.GetOptions{})
+				if err != nil {
+					return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 128) but it returns nil

(nilerr)